src/eosjs-webauthn-sig.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @module WebAuthn-Sig
+ */
+// copyright defined in eosjs/LICENSE.txt
+
+import { SignatureProvider, SignatureProviderArgs } from './eosjs-api-interfaces';
+import * as ser from './eosjs-serialize';
+import * as numeric from './eosjs-numeric';
+import { ec } from 'elliptic';
+
+/** Signs transactions using WebAuthn */
+export class WebAuthnSignatureProvider implements SignatureProvider {
+    /** Map public key to credential ID (hex). User must populate this. */
+    public keys = new Map<string, string>();
+
+    /** Public keys that the `SignatureProvider` holds */
+    public async getAvailableKeys() {
+        return Array.from(this.keys.keys());
+    }
+
+    /** Sign a transaction */
+    public async sign(
+        { chainId, requiredKeys, serializedTransaction, serializedContextFreeData }:
+            SignatureProviderArgs,
+    ) {
+        const signBuf = new ser.SerialBuffer();
+        signBuf.pushArray(ser.hexToUint8Array(chainId));
+        signBuf.pushArray(serializedTransaction);
+        if (serializedContextFreeData) {
+            signBuf.pushArray(new Uint8Array(await crypto.subtle.digest('SHA-256', serializedContextFreeData.buffer)));
```

Comment:
Using `crypto.subtle` will break Nodejs compatibility, since Node doesn't support the WebCrypto API fully.

Reply:
nodejs can't possibly support webauthn. It's a browser-only standard.

Reply:
I had that thought after I typed, see below comment. We should at least note in the README/Documentation somehow that this SigProvider is browser only